docs/how-to/react-server-components.md

```diff
@@ -1,50 +1,738 @@
+
+**Server**
+
+<docs-info>
+
+You don't have to use SSR at all. You can choose to use RSC to "prerender" HTML for Static Site Generation (SSG) or something like Incremental Static Regeneration (ISR).
+
+</docs-info>
+
+`entry.ssr.tsx` is the entry point for the server. It is responsible for handling the request, calling the RSC server, and converting the RSC payload into HTML on document requests (server-side rendering).
+
+Relevant APIs:
+
+- [`routeRSCServerRequest`][route-rsc-server-request]
+- [`RSCStaticRouter`][rsc-static-router]
+
+**RSC Server**
+
+<docs-info>
+
+Even though you have a "React Server" and a server responsible for request handling/SSR, you don't actually need to have 2 separate servers. You can simply have 2 separate module graphs within the same server. This is important because React behaves differently when generating RSC payloads vs. when generating HTML to be hydrated on the client.
+
+</docs-info>
+
+`entry.rsc.tsx` is the entry point for the React Server. It is responsible for matching the request to a route and generating RSC payloads.
+
+Relevant APIs:
+
+- [`matchRSCServerRequest`][match-rsc-server-request]
+
+**Client**
+
+`entry.client.tsx` is the entry point for the client. It is responsible for hydrating the generated HTML and setting the `callServer` function to support post-hydration server actions.
+
+Relevant APIs:
+
+- [`createCallServer`][create-call-server]
+- [`getRSCStream`][get-rsc-stream]
+- [`RSCHydratedRouter`][rsc-hydrated-router]
+
+### Parcel
+
+See the [Parcel RSC docs][parcel-rsc-doc] for more information.
+
+In addition to `react`, `react-dom`, and `react-router`, you'll need the following dependencies:
+
+```shellscript
+# install runtime dependencies
+npm i @parcel/runtime-rsc react-server-dom-parcel
+
+# install dev dependencies
+npm i -D parcel
+```
+
+#### `package.json`
+
+To configure Parcel, add the following to your `package.json`:
+
+```json filename=package.json
+{
+  "scripts": {
+    "build": "parcel build --no-autoinstall",
+    "dev": "cross-env NODE_ENV=development parcel --no-autoinstall --no-cache",
+    "start": "cross-env NODE_ENV=production node dist/server/entry.rsc.js"
+  },
+  "targets": {
+    "react-server": {
+      "context": "react-server",
+      "source": "src/entry.rsc.tsx",
+      "scopeHoist": false,
+      "includeNodeModules": {
+        "@mjackson/node-fetch-server": false,
+        "compression": false,
+        "express": false
+      }
+    }
+  }
+}
+```
+
+#### `routes/config.ts`
+
+You must add `"use server-entry"` to the top of the file where you define your routes. Additionally, you need to import the client entry point, since it will use the `"use client-entry"` directive (see below).
+
+```tsx filename=src/routes/config.ts
+"use server-entry";
+
+import type { unstable_RSCRouteConfig as RSCRouteConfig } from "react-router";
+
+import "../entry.client";
+
+export function routes() {
```

Comment:
It might be worth calling out that this needs to be a function so that Parcel adds a `bootstrapScript` property, otherwise this function looks redundant.